provider/image.go

```diff
@@ -93,13 +95,29 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 	// make the build context and ensure to exclude dockerignore file patterns
 	dockerIgnorePath := filepath.Join(build.Context, ".dockerignore")
 	ignorePatterns, err := getIgnore(dockerIgnorePath)
+	ignorePatterns = buildCmd.TrimBuildFilesFromExcludes(ignorePatterns, img.Build.Dockerfile, false)
+
+	// warn user about accidentally copying build files
+	if build.BuilderVersion == defaultBuilder {
+		for _, pattern := range ignorePatterns {
+			if pattern == "!Dockerfile" || pattern == "!.Dockerignore" {
+				msg := "It looks like you are trying to dockerignore a build file such as Dockerfile or .dockerignore. " +
+					"Due to limitations when running this provider in Buildkit mode, your build files may get copied " +
+					"into your image. Please ensure any copied file systems do not include build files."
```

Comment:
This only applies to .dockerignore and the Dockerfile. Perhaps we should tone this message down, that seems very low impact.

Reply:
how is this?